apps/beeai-server/src/beeai_server/api/dependencies.py

```diff
@@ -35,19 +44,69 @@
 UserServiceDependency = Annotated[UserService, Depends(lambda: di[UserService])]
 VectorStoreServiceDependency = Annotated[VectorStoreService, Depends(lambda: di[VectorStoreService])]
 UserFeedbackServiceDependency = Annotated[UserFeedbackService, Depends(lambda: di[UserFeedbackService])]
+AuthServiceDependency = Annotated[AuthService, Depends(lambda: di[AuthService])]
 
 logger = logging.getLogger(__name__)
+api_key_cookie = APIKeyCookie(name="beeai-platform", auto_error=False)
+
+
+async def authenticate_oauth_user(
+    bearer_auth: HTTPAuthorizationCredentials,
+    cookie_auth: str | None,
+    user_service: UserServiceDependency,
+    configuration: ConfigurationDependency,
+) -> AuthorizedUser:
+    """
+    Authenticate using an OIDC/OAuth2 JWT bearer token with JWKS.
+    Creates the user if it doesn't exist.
+    """
+    try:
+        token = extract_oauth_token(bearer_auth, cookie_auth)
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail=f"Invalid Authorization header: {e}",
+        ) from e
+
+    claims, issuer = await decode_oauth_jwt_or_introspect(
+        token, jwks_dict=di[JWKS_DICT], aud="beeai-server", configuration=configuration
+    )
+    if not claims:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid or expired token")
+
+    email = claims.get("email")
+    if not email:
+        provider = next((p for p in configuration.auth.oidc.providers if p.issuer == issuer), None)
+        if not provider:
+            raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="issuer not configured")
+        userinfo = await fetch_user_info(token, f"{provider.issuer}/userinfo")
+        email = userinfo.get("email")
+
+    if not email:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Email not available in token or userinfo")
+
+    is_admin = email in configuration.auth.oidc.admin_emails
+
+    try:
+        user = await user_service.get_user_by_email(email=email)
+    except EntityNotFoundError:
+        role = UserRole.admin if is_admin else UserRole.user
+        user = await user_service.create_user(email=email, role=role)
+
+    return AuthorizedUser(
+        user=user,
+        global_permissions=ROLE_PERMISSIONS[user.role],
+        context_permissions=ROLE_PERMISSIONS[user.role],
+    )
 
 
 async def authorized_user(
     user_service: UserServiceDependency,
     configuration: ConfigurationDependency,
     basic_auth: Annotated[HTTPBasicCredentials | None, Depends(HTTPBasic(auto_error=False))],
     bearer_auth: Annotated[HTTPAuthorizationCredentials | None, Depends(HTTPBearer(auto_error=False))],
+    cookie_auth: Annotated[str | None, Security(api_key_cookie)],
```

Comment:
what is the `cookie_auth` is it still a leftover from a session middleware or do we need this somewhere?